src/connector/sql/sqlGeneration.ts

```diff
@@ -233,57 +317,115 @@ export const scalarComparisonOperatorMappings: ScalarOperatorMappings = {
   },
   String: {
     comparison: {
+      is_null: {
+        name: "IS_NULL",
+        isInfix: false,
+        isUnary: true,
+      },
+
       eq: {
         name: "=",
         isInfix: true,
+        isUnary: false,
       },
       neq: {
         name: "!=",
         isInfix: true,
+        isUnary: false,
       },
       gt: {
         name: ">",
         isInfix: true,
+        isUnary: false,
       },
       lt: {
         name: "<",
         isInfix: true,
+        isUnary: false,
       },
       gte: {
         name: ">=",
         isInfix: true,
+        isUnary: false,
       },
       lte: {
         name: "<=",
         isInfix: true,
+        isUnary: false,
       },
       contains: {
         name: "CONTAINS",
         isInfix: false,
+        isUnary: false,
       },
       endswith: {
         name: "ENDSWITH",
         isInfix: false,
+        isUnary: false,
       },
       regexmatch: {
         name: "REGEXMATCH",
         isInfix: false,
+        isUnary: false,
       },
       startswith: {
         name: "STARTSWITH",
         isInfix: false,
+        isUnary: false,
       },
     },
   },
 };
 
+export function getScalarType(column: Column): string {
+  if (column.nestedField) {
+    return getNestedScalarType(column.nestedField);
+  } else {
+    if (column.type === undefined) {
+      throw new sdk.BadRequest(`Couldn't find type of column: ${column.name}`);
+    } else {
+      switch (column.type.type) {
+        case "array":
+          throw new sdk.BadRequest(
+            `Expected column ${column.name} to be a scalar type, but found array type`,
+          );
+        case "named":
+          if (column.type.kind === "object") {
+            throw new sdk.BadRequest(
+              `Expected column ${column.name} to be a scalar type, but found object type`,
+            );
+          } else {
+            return column.type.name;
+          }
+        case "nullable":
+          return getScalarType({
+            name: column.name,
+            prefix: column.prefix,
+            type: column.type.underlyingType,
+          });
+      }
+    }
+  }
+}
+
+function getNestedScalarType(nestedField: NestedField): string {
```

Comment:
I don't understand this function well, could you please explain it. What I'm reading here is:
For nestedField of type object/array, we always assume the underlying type to be a scalar, and call the getNestedScalarType function. I'm not clear on why we are assuming that

Reply:
> For nestedField of type object/array, we always assume the underlying type to be a scalar, and call the getNestedScalarType function. 

Yes, we first validate that the last element of the nested field should be a scalar because you can apply a binary/unary comparison operator value only on a scalar value. So, if we don't find a scalar type at the end, the request is deemed as an invalid request.